shell/packages/sandstorm-permissions/permissions.js

```diff
@@ -291,100 +252,173 @@ class RequirementSet {
       delete this.identityPermissions[grainId];
     }
 
-    return { vertex: { grain: { _id: grainId, identityId: identityId } },
-             permissions: permissions, };
+    return { grainId: grainId, identityId: identityId, permissionSet: permissions, };
+  }
+}
+
+const permissionIdPattern = Match.OneOf({ canAccess: null }, { appDefined: Match.Integer });
```

Comment:
I've switched to a plain object. Now a permission ID is a stringified integer >= -1, where "-1" means "can access".  This does simplify a few things!